lib/internal/http2/core.js

```diff
@@ -2731,11 +2730,9 @@ function connect(authority, options, listener) {
 // Support util.promisify
 Object.defineProperty(connect, promisify.custom, {
   value: (authority, options) => {
-    const promise = createPromise();
-    const server = connect(authority,
-                           options,
-                           () => promiseResolve(promise, server));
-    return promise;
+    return new Promise((resolve) => {
+      const server = connect(authority, options, () => resolve(server));
```

Comment:
This changes existing behavior - if an invalid URL is passed as an authority [here](https://github.com/BridgeAR/node/blob/be458a82ee51f85e552faa49f4938d617464c4e9/lib/internal/http2/core.js#L2692) then the function throws. The old behavior was to throw synchronously but the new behavior throws asynchronously. 

In my opinion the new behavior is a behavior change in error messages though. I'm not sure how it can be resolved in the promise constructor or done without reverting the change. 

What do you think?

Reply:
Honestly I think it's best to handle it async too but I'm not sure I get to make that call. Pinging previous approvers for more OKs - @addaleax @bmeurer @TimothyGu @devsnek - please 👍 this comment to agree to landing this (arguably bugfix) behaviour change.

Reply:
I think this is a breaking change – otherwise no strong opinions.

Reply:
@addaleax I do not see how someone would really rely on this.

Most people will probably not wrap a promise call in a try catch. So it'll end up as uncaught exception and that will likely end the process. That is not really something that anyone would want.

What scenario do you see that would break with this change?

Reply:
I definitely believe this is something that was likely overlooked when implementing it and I feel strongly that async functions should not throw synchronously. 

That said, this _is_ a breaking change in behaviour and would require appropriate care. An example scenario that could happen is:

```js
function connectServer(options) {
  try {
    // in the actual case - a legitimate invalid URL from the user
    return connect(env.SERVER_AUTHORITY, options); 
  } catch (ex) {
    // handle malformed url here, for example by connecting to the dev-server
    // or by showing a useful error - this will stop working with this PR.
  }
}
```

Now, I don't think it's a _good_ idea for developer happiness to write code like the above - but I can see cases where people would write code like this. I think the breakage _might_ be worth it in this case and I like the new behavior better - but I'd like the TSC to weigh in if we want to land it as semver-patch.

Reply:
@nodejs/tsc PTAL

Reply:
@BridgeAR What are the benefits of not considering it semver-major? I see a very valid point in this adding consistency, yes. We’ve definitely treated error timing changes in the past as `semver-major`, but I won’t stand in the way of making an exception here.

How important is this? Do you want to have this patch in 10.x for another reason besides the behavioral improvement here?

Reply:
This is a change in experimental http2 code. There no reason to make it semver-major on the basis of this one change.

Reply:
Oh, right, http2 is still experimental. Fine with me then. :)

(I had originally added the `dont-land-on-v10.x` label because of perf concerns anyway – feel free to remove it if you like.)

Reply:
I did not realize http2 is still experimental either - I'm fine with semver-patch :)

Reply:
@addaleax 
> How important is this? Do you want to have this patch in 10.x for another reason besides the behavioral improvement here?

Backporting is easier as well and users get the better behavior earlier :-) (even though that only makes sense for 10.).

I do not have a strong opinion on it though. If you want, I can change the timers and the child_process to behave exactly as before but I personally still believe this is a bug fix as it was never meant to behave like that (and it only did that in the first place because of the special implementation before).

Reply:
@BridgeAR IIUC @addaleax is fine landing this as semver patch https://github.com/nodejs/node/pull/20830#discussion_r191059143 and later added:

> (I had originally added the dont-land-on-v10.x label because of perf concerns anyway – feel free to remove it if you like.)

Which means that this is fine to land as semver patch and on v10.x